doc/glossary.rst

```diff
@@ -127,6 +135,16 @@ Glossary of Terms
       Power gating reduces power consumption by shutting off areas of an
       integrated circuit that are not in use.
 
+   private API
+      Any private function, structure or macro defined anywhere in the zephyr source
+      tree. Private APIs are only intended for consumption inside the
+      :term:`software component` where they are defined.
+
+   public API
+      Any public function, structure or macro defined inside the ``include/zephyr``
+      folder. Public APIs are intended for consumption by any and all in-tree or
+      out-of-tree :term:`software components <software component>`.
```

Comment:
Following up from @henrikbrixandersen comment at arch wg: there's few subsystems that use "common" struct in their driver definitions (gpio, can, regulator, input kbd...). These are often defined in the same file as the proper application API, for instance `include/zephyr/drivers/gpio.h` has both `struct gpio_driver_config` and `gpio_pin_set_dt`.

They could be used for an out-of-tree driver, but it seems like they should belong to different categories stability wise, so putting everything in `include/zephyr` under the same umbrella seems excessive. Is the expectations that such APIs/structs gets the `_priv_` infix?

Reply:
I agree.
As a side comment, code organization wise, maybe those APIs should not be in the same header as the app facing API. (sometimes we put them in 1 file, just because it is easy).

Reply:
Yeah it feels like we are moving in the direction of a header split, something like the Linux uapi one.